src/azure-cli/azure/cli/command_modules/sql/_params.py

```diff
@@ -959,6 +960,57 @@ def _configure_security_policy_storage_params(arg_ctx):
         c.argument('allow_data_loss',
                    arg_type=allow_data_loss_param_type)
 
+    ###############################################
+    #             sql instance pool               #
+    ###############################################
+
+    with self.argument_context('sql instance-pool') as c:
+        c.argument('instance_pool_name', options_list=['--name', '-n'],
+                   help="Instance Pool Name")
+        c.argument(
+            'tier',
+            arg_type=tier_param_type,
+            required=True,
+            help='The edition component of the sku. Allowed values: GeneralPurpose, BusinessCritical.')
+
+        c.argument('family',
+                   arg_type=family_param_type,
+                   required=True,
+                   help='The compute generation component of the sku. '
+                   'Allowed values include: Gen4, Gen5.')
+
+    with self.argument_context('sql instance-pool create') as c:
+        # Create args that will be used to build up the InstancePool object
+        create_args_for_complex_type(
+            c, 'parameters', InstancePool, [
+                'location',
+                'license_type',
+                'subnet_id',
+                'vcores',
+                'tags'
+            ])
+
+        c.argument('vcores',
+                   required=True,
+                   arg_type=capacity_param_type,
+                   help='the capacity of the instance pool in vcores.')
+
+        c.argument(
+            'subnet_id',
```

Comment:
As @jaredmoo said, I think we can both support specifying subnet using subnet_id or subnet_name&vnet_name.
When users use  subnet_name&vnet_name, subnet should be in the same resource group of instance pool.
But @rohands  out of curiosity, why resource group is not required for your commands?

Reply:
Resource group is required for the commands. I've added the functionality to support adding vnets and subnet name. Please take a look at validators.py